src/main/java/seedu/address/model/transaction/TransactionLog.java

```diff
@@ -79,6 +76,38 @@ public ObservableList<Transaction> asUnmodifiableObservableList() {
         return FXCollections.unmodifiableObservableList(internalList);
     }
 
+    /**
+     * @return the number of transactions in the transaction log.
+     */
+    public int size() {
+        return transactionList.size();
+    }
+
+    /**
+     * Retrieves a transaction at specified index.
+     *
+     * @param index of transaction to be retrieved.
+     * @return transaction at the index
+     */
+    public Transaction getTransaction(int index) {
+        return transactionList.get(index);
+    }
+
+    /**
+     * Sets a new transaction at the specified index of transaction list.
+     *
+     * @param index of transaction to be replaced.
+     * @param editedTransaction the transaction to be used to replace the old transaction at the specified index.
+     */
+    public void setTransaction(int index, Transaction editedTransaction) {
+        requireAllNonNull(editedTransaction);
+
+        if (index == -1) {
+            throw new TransactionNotFoundException();
```

Comment:
Perhaps could add this to the test case?

Reply:
Thanks for suggesting this! Ive added a test case